radixdlt/src/test/java/com/radixdlt/sanitytestsuite/scenario/SanityTestScenarioRunner.java

```diff
@@ -47,14 +43,29 @@ public void executeTest(SanityTestSuiteRoot.SanityTestSuite.SanityTestScenario s
 
 		for (int testVectorIndex = 0; testVectorIndex < scenario.tests.vectors.size(); ++testVectorIndex) {
 			UnknownTestVector untypedVector = scenario.tests.vectors.get(testVectorIndex);
-			TestVector testVector = cast(untypedVector, this.typeOfVector());
+
+			ObjectMapper mapper = new ObjectMapper();
+			String unknownTestVectorJSONString = null;
+			try {
+				unknownTestVectorJSONString = mapper.writeValueAsString(untypedVector);
```

Comment:
Uh... ugly! How can we avoid this?! Here I'm converting `UnknownTestVector` back into JSON string so that I can (a couple of lines below) cast it into `TestVector` (Generic type for subclasses of this class) using 

```java
mapper.readValue(unknownTestVectorJSONString, this.testVectorTypeReference());
```

The `ObjectMapper` did not have any method `readValue(Object, TypeReference)` signature, only JSON **String**.

Surely there must be a MUCH simpler way of casting from `UnknownTestVector` to the generic type `TestVector` directly?!